plugin/src/main/java/org/opennms/velocloud/rest/api/AbstractVelocloudRestService.java

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2022 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2022 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+package org.opennms.velocloud.rest.api;
+
+import org.opennms.velocloud.client.api.VelocloudApiClient;
+import org.opennms.velocloud.client.api.VelocloudApiException;
+import org.opennms.velocloud.connections.ConnectionManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public abstract class AbstractVelocloudRestService implements VelocloudRestService {
+
+    private final ConnectionManager connectionManager;
+
+    private static final Logger LOG = LoggerFactory.getLogger(AbstractVelocloudRestService.class);
+
+    public AbstractVelocloudRestService(final ConnectionManager connectionManager) {
+        this.connectionManager = connectionManager;
+    }
+
+    public VelocloudApiClient getClientFor(final String alias) throws VelocloudApiException {
+        return this.connectionManager.getClientFor(alias);
```

Comment:
If this would return an `Optional<VelocloudApiClient` we can throw a not-found-exception with the right message here.